### ISSUE TITLE
Digest Auth client middleware

### ISSUE THREAD

**Issue by kuzkdmy** [http4s/http4s#3588]
Http4s provide Digest auth middleware to implement server, but luck of client middleware
This can be really not so trivial task for somebody who will just start working with http4s. 
I have this already being implemented, need to revisit and contribute this to become a part of http4s project

**Comment by Daenyth** [comment]
To make sure it's clear: this is a request to add a client middleware for `Digest` auth, similar to the server middleware that exists now?

**Comment by kuzkdmy** [comment]
Yes, sorry that don't find time to make a pull request. I'll have this already being implemented, just need to revisit and make a pull request, will try to make this week.

**Comment by rossabaker** [comment]
@kuzkdmy Great! It sounds like this is new and wouldn't change any internals, so if you target the series/0.21 branch, we can get it into the production release series sooner.

**Comment by kuzkdmy** [comment]
Migrate from what we used with a very little refactoring
https://github.com/http4s/http4s/pull/3618/files
Thanks
